packages/otelbin/src/app/social-preview/[id]/page.tsx

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Redis } from "@upstash/redis/nodejs";
+import type { Metadata } from "next";
+import type { IConfig } from "~/components/react-flow/dataType";
+import { extractComponents, parseUrlFragment, sortAndDeduplicate } from "~/lib/metadataUtils";
+import { getShortLinkPersistenceKey } from "~/lib/shortLink";
+
+interface ExtendedMetadata {
+	twitterData1: string;
+	twitterLabel1: string;
+	twitterData2: string;
+	twitterLabel2: string;
+}
+
+const redis = Redis.fromEnv();
+const width = 1200;
+const height = 630;
+const ogImageAlt = "OpenTelemetry collector configuration pipeline visualization by OTelBin";
+
+export async function generateMetadata({ params }: { params: { id: string } }): Promise<Metadata> {
+	const extendedMetadata: ExtendedMetadata = {
+		twitterLabel1: "Components",
+		twitterData1: "",
+		twitterLabel2: "Pipelines",
+		twitterData2: "",
+	};
+	const fullLink = (await redis.get<string>(getShortLinkPersistenceKey(params.id))) ?? "";
```

Comment:
**Question:** What if the ID doesn't resolve to any config? 

It looks like we are missing error handling here.